migrations/20221012165909-add-stripe-charge-id-index.js

```diff
@@ -0,0 +1,14 @@
+'use strict';
+
+module.exports = {
+  async up(queryInterface) {
+    await queryInterface.sequelize.query(`
+      CREATE INDEX CONCURRENTLY stripe_charge_id 
+      ON "Transactions" (((data -> 'charge' ->> 'id')::text)) 
+      WHERE (data -> 'charge' ->> 'id') IS NOT NULL`);
```

Comment:
Have you tested this index? I think I had some trouble in the past related to the JSONB selection operator needing to match the operator used in the query.
I'm not 100% confident here but I implemented something similar this week that I can ensure would work:
```suggestion
      CREATE INDEX CONCURRENTLY IF NOT EXISTS "transactions__stripe_charge_id" ON "Transactions" USING BTREE ((data#>>'{charge,id}') DESC) WHERE data#>>'{charge,id}' IS NOT NULL;
```

Reply:
Here is the same query after applying the index:
```
opencollective_prod_snapshot> explain analyze select id from "Transactions" t where t.data->'charge'->>'id' = 'ch_3L4DagBYycQg1OMf1KxilamS';
+-------------------------------------------------------------------------------------------------------------------------------+
| QUERY PLAN                                                                                                                    |
|-------------------------------------------------------------------------------------------------------------------------------|
| Bitmap Heap Scan on "Transactions" t  (cost=387.08..48602.90 rows=16342 width=4) (actual time=0.069..0.074 rows=2 loops=1)    |
|   Recheck Cond: (((data -> 'charge'::text) ->> 'id'::text) = 'ch_3L4DagBYycQg1OMf1KxilamS'::text)                             |
|   Heap Blocks: exact=2                                                                                                        |
|   ->  Bitmap Index Scan on stripe_charge_id  (cost=0.00..382.99 rows=16342 width=0) (actual time=0.055..0.056 rows=2 loops=1) |
|         Index Cond: (((data -> 'charge'::text) ->> 'id'::text) = 'ch_3L4DagBYycQg1OMf1KxilamS'::text)                         |
| Planning Time: 0.230 ms                                                                                                       |
| Execution Time: 0.139 ms                                                                                                      |
+-------------------------------------------------------------------------------------------------------------------------------+
EXPLAIN
Time: 0.018s
```

Reply:
You'll need to analyze the query that is generated by:
```js
const transactions = await models.Transaction.findAll({
    where: { data: { charge: { id: stripeChargeId } } }
});
```

Reply:
Ah ok, I can see that the query generated from the `findAll` is using the other query operator `"Transaction"."data"#>>'{charge,id}'` and when I ran that query against the prod dump it was about 10 seconds. I applied the index you posted above and it improved dramatically again. 

```
opencollective_prod_snapshot> SELECT id  FROM "Transactions" WHERE "data"#>>'{charge,id}' = 'ch_3L4DagBYycQg1OMf1KxilamS';
+---------+
| id      |
|---------|
| 2495478 |
| 2495479 |
+---------+
SELECT 2
Time: 9.967s (9 seconds), executed in: 9.959s (9 seconds)
opencollective_prod_snapshot> CREATE INDEX CONCURRENTLY IF NOT EXISTS "transactions__stripe_charge_id" ON "Transactions" USING BTREE ((data#>>'{cha
 rge,id}') DESC) WHERE data#>>'{charge,id}' IS NOT NULL;
CREATE INDEX
Time: 31.947s (31 seconds), executed in: 31.947s (31 seconds)
opencollective_prod_snapshot> SELECT id  FROM "Transactions" WHERE "data"#>>'{charge,id}' = 'ch_3L4DagBYycQg1OMf1KxilamS';
+---------+
| id      |
|---------|
| 2495478 |
| 2495479 |
+---------+
SELECT 2
Time: 0.021s
```

Reply:
Would it be worth it to create both indices? Just in case some tries to use this query operator `(data -> 'charge'::text) ->> 'id'::text`?

Reply:
```
opencollective_prod_snapshot> explain analyze SELECT id  FROM "Transactions" WHERE "data"#>>'{charge,id}' = 'ch_3L4DagBYycQg1OMf1KxilamS';
+---------------------------------------------------------------------------------------------------------------------------------------------+
| QUERY PLAN                                                                                                                                  |
|---------------------------------------------------------------------------------------------------------------------------------------------|
| Bitmap Heap Scan on "Transactions"  (cost=387.08..48562.05 rows=16342 width=4) (actual time=0.060..0.066 rows=2 loops=1)                    |
|   Recheck Cond: ((data #>> '{charge,id}'::text[]) = 'ch_3L4DagBYycQg1OMf1KxilamS'::text)                                                    |
|   Heap Blocks: exact=2                                                                                                                      |
|   ->  Bitmap Index Scan on transactions__stripe_charge_id  (cost=0.00..382.99 rows=16342 width=0) (actual time=0.048..0.048 rows=2 loops=1) |
|         Index Cond: ((data #>> '{charge,id}'::text[]) = 'ch_3L4DagBYycQg1OMf1KxilamS'::text)                                                |
| Planning Time: 0.218 ms                                                                                                                     |
| Execution Time: 0.144 ms                                                                                                                    |
+---------------------------------------------------------------------------------------------------------------------------------------------+
EXPLAIN
Time: 0.016s
```

Reply:
Resolved the final question off-thread: just going to create an index that matches the operator style used by Sequelize querying.